### PR TITLE
Load/Save campaign heroes between scenarios

### PR DESCRIPTION
**Pull request by Kert**
It's possible now to control which hero data transfers to the next scenario by adding a few lines to a corresponding campaign file (*.cmp)
Data transfered:
- Primary/secondary skills
- Spells learned
- Experience

Examples:
`<saveHero scenarioID="0" playerID="0" ownedHeroID="0"/>`
This will save the first owned hero from player 0 to memory on the first campaign scenario

`<loadHero scenarioID="1" playerID="0" ownedHeroID="0"/>`
This will load the first saved hero in memory to the first hero owned by player 0

It's possible to transfer multiple heroes between scenarios but you must follow the same order for saving and loading, e.g.:
`<saveHero scenarioID="0" playerID="0" ownedHeroID="0"/>`
`<saveHero scenarioID="0" playerID="0" ownedHeroID="1"/>`
`<loadHero scenarioID="1" playerID="0" ownedHeroID="0"/>`
`<loadHero scenarioID="1" playerID="0" ownedHeroID="1"/>`

### CODE REVIEW

**Comment by jkoppel** [comment]
Nice!

Can I see a video of this?

File: src/cpp/shared/expansions.h [hunk 1]
```diff
@@ -12,7 +12,26 @@ class combatExtra {
         stackExtra stack;
 };
 
+class campaignExtra {
```
Comment (jkoppel) [inline src/cpp/shared/expansions.h]: This is a brand new class, not present in the original game, right?

**Comment by jkoppel** [comment]
I approve this pending seeing it run, and pending upstream branches.

File: src/cpp/shared/expansions.h (hunk 1 above)
Comment (Kert) [reply src/cpp/shared/expansions.h]: Yes. It's a new one

**Pull request merged**